src/main/java/org/fcrepo/client/FcrepoLink.java

```diff
@@ -58,41 +82,81 @@ public URI getUri() {
      * @return the "rel" portion of a Link header
      */
     public String getRel() {
-        return rel;
+        return getParam(META_REL);
+    }
+
+    /**
+     * Retrieve the type portion of the link
+     *
+     * @return the "type" parameter of the header
+     */
+    public String getType() {
+        return getParam(META_TYPE);
+    }
+
+    /**
+     * Retrieve a parameter from the link header
+     *
+     * @param name name of the parameter in the link header
+     * @return the value of the parameter or null if not present.
+     */
+    public String getParam(final String name) {
+        return params.get(name);
+    }
+
+    /**
+     * Retrieve a map of parameters from the link header
+     *
+     * @return map of parameters
+     */
+    public Map<String, String> getParams() {
+        return params;
     }
 
     /**
      * Parse the value of a link header
      */
     private void parse(final String link) {
-        if (link != null) {
-            final String[] segments = link.split(LINK_DELIM);
+        final String[] segments = link.split(LINK_DELIM, 2);
+        if (segments.length > 0) {
+            uri = getLinkPart(segments[0]);
+            // Parse the remainder of the header after the first ; as parameters
             if (segments.length > 1) {
-                uri = getLinkPart(segments[0]);
-                if (uri != null) {
-                    // inspect the remaining segments until a rel is found
-                    for (int i = 1; i < segments.length && rel == null; i++) {
-                        rel = getRelPart(segments[i]);
-                    }
-                }
+                parseParams(segments[1]);
             }
         }
     }
 
-    /**
-     * Extract the rel="..." part of the link header
-     */
-    private static String getRelPart(final String relPart) {
-        final String[] segments = relPart.trim().split("=");
-        if (segments.length != 2 || !META_REL.equals(segments[0])) {
-            return null;
+    private void parseParams(final String paramString) {
+        final Matcher paramMatcher = LINK_PARAM_PATTERN.matcher(paramString);
+        while (paramMatcher.find()) {
+            final String name = paramMatcher.group(1);
+            // May only specify parameters once, subsequent cases of the same param must be ignored
+            if (params.containsKey(name)) {
+                continue;
+            }
+
+            final String value = stripQuotes(paramMatcher.group(2));
+            params.put(name, value);
```

Comment:
This Link class does not currently support, legal, multi-value Link headers, such as the one found in the Memento spec ([4.2.1](https://tools.ietf.org/html/rfc7089#section-4.2.1))
```
        final FcrepoLink l = new FcrepoLink("<http://a.example.org/>; rel=\"original\", "+
    "<http://arxiv.example.net/timemap/http://a.example.org/>" +
        "; rel=\"timemap\"; type=\"application/link-format\"" +
        "; from=\"Tue, 15 Sep 2000 11:28:26 GMT\"" +
        "; until=\"Wed, 20 Jan 2010 09:34:33 GMT\"");
```

Reply:
I opted not to address this case, leaving it up to the client to do the splitting into multiple headers before deserializing into FcrepoLink objects. I don't mind adding this, but it will require a separate method which returns `List<FcrepoLink>`.
For what its worth, as far as I can tell the jersey implementation doesn't support this case either. If I run:
```
final Link link = Link.valueOf("<http://a.example.org/>; rel=\"original\", " +
                "<http://arxiv.example.net/timemap/http://a.example.org/>" +
                "; rel=\"timemap\"; type=\"application/link-format\"" +
                "; from=\"Tue, 15 Sep 2000 11:28:26 GMT\"" +
                "; until=\"Wed, 20 Jan 2010 09:34:33 GMT\"");
```
I get
```
java.lang.IllegalArgumentException: Unable to parse link <http://a.example.org/>; rel="original", <http://arxiv.example.net/timemap/http://a.example.org/>; rel="timemap"; type="application/link-format"; from="Tue, 15 Sep 2000 11:28:26 GMT"; until="Wed, 20 Jan 2010 09:34:33 GMT"
	at org.glassfish.jersey.message.internal.LinkProvider.initBuilder(LinkProvider.java:125)
	at org.glassfish.jersey.message.internal.JerseyLink$Builder.link(JerseyLink.java:97)
	at org.glassfish.jersey.message.internal.JerseyLink$Builder.link(JerseyLink.java:73)
	at javax.ws.rs.core.Link.valueOf(Link.java:173)
```

Reply:
Maybe an exception would be a good solution in our implementation as well.